packages/api/src/routes/profileRoutes.js

```diff
@@ -47,41 +55,13 @@ profileRoutes.post('/createUser', isAuthenticated, (req, res) => {
   if (body.role === 'student') {
     body.batchId = body.batchId.toString();
   }
-  const user = new User(body);
-  return user.save()
-    .then((doc) => {
-      if (!doc || doc.length === 0) {
-        return res.status(500).json({ error: ERR_MSGS.internalServerError });
-      }
-      return res.status(201).json({ user_created: 'Success', user_email: doc.email });
-    })
-    .catch(() => {
-      res.status(500).json({ error: ERR_MSGS.internalServerError });
-    });
-});
-/* Update request accepts whole body but should contain only the fields in User schema
-* The input fields are meant to be disabled on front end such as city, batchNumber, email
-*/
-profileRoutes.put('/user/:id', isAuthenticated, extractUser, async (req, res) => {
-  const { id } = req.params;
-  const { user } = req;
-  if (!user) {
-    res.status(400).json({ error: ERR_MSGS.profileNotExist });
-  }
-  if (id.toLowerCase() === 'me') {
-    await User.findOneAndUpdate(
-      { email: user.email },
-      req.body,
-      { new: true, runValidators: true },
-      (err, newUserDetails) => {
-        if (err) {
-          res.status(500).json({ error: ERR_MSGS.internalServerError });
-        }
-        return res.json(newUserDetails);
-      },
-    );
+  if (!req.user) {
+    const user = new User(body);
+    const result = await user.save();
+    return res.send(result);
   }
-  res.status(400).end();
+  const result = await User.updateOne({ email: req.user.email }, body, { upsert: true });
+  return res.send(result);
```

Comment:
The result returned by mongoose is not meaningful to the frontend dev. Just do res.json({success: true})